enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/CustomProbeEnricher.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright 2016 Red Hat, Inc.
+ *
+ * Red Hat licenses this file to you under the Apache License, version
+ * 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package io.fabric8.maven.enricher.standard;
+
+import io.fabric8.kubernetes.api.builder.TypedVisitor;
+import io.fabric8.kubernetes.api.model.ContainerBuilder;
+import io.fabric8.kubernetes.api.model.KubernetesListBuilder;
+import io.fabric8.kubernetes.api.model.Probe;
+import io.fabric8.maven.core.config.ProbeConfig;
+import io.fabric8.maven.core.config.ResourceConfig;
+import io.fabric8.maven.core.handler.ProbeHandler;
+import io.fabric8.maven.enricher.api.BaseEnricher;
+import io.fabric8.maven.enricher.api.MavenEnricherContext;
+
+import java.util.NoSuchElementException;
+
+public class CustomProbeEnricher extends BaseEnricher {
+    public CustomProbeEnricher(MavenEnricherContext buildContext) {
+        super(buildContext, "fmp-customprobe");
+    }
+
+    @Override
+    public void addMissingResources(KubernetesListBuilder builder) {
+        builder.accept(new TypedVisitor<ContainerBuilder>() {
+            @Override
+            public void visit(ContainerBuilder containerBuilder) {
+                try {
+                    ProbeHandler probeHandler = new ProbeHandler();
+                    ResourceConfig resourceConfig = getConfiguration().getResource().get();
+
+                    ProbeConfig readinessProbeConfig = resourceConfig.getReadiness();
+                    if (readinessProbeConfig != null) {
+                        Probe readinessProbe = probeHandler.getProbe(readinessProbeConfig);
+                        if (readinessProbe != null) {
+                            containerBuilder.withReadinessProbe(readinessProbe);
+                        }
+                    }
+
+                    ProbeConfig livenessProbeConfig = resourceConfig.getLiveness();
+                    if (livenessProbeConfig != null) {
+                        Probe livenessProbe = probeHandler.getProbe(livenessProbeConfig);
+                        if (livenessProbe != null) {
+                            containerBuilder.withLivenessProbe(livenessProbe);
+                        }
+                    }
```

Comment:
I think we could create a method which returns the `Probe`.

```java
getProbe(ProbeConfig pc) {
  if (pc != null) {
    return  probeHandler.getProbe(pc);
  }
  return null;
}
```

And then it is called twice:

```java
containerBuilder.withLivenesssProve(getProbe(resourceConfig.getLiveness())
```